discretisedfield/field.py

```diff
@@ -272,29 +259,23 @@ def value(self):
         ...
         >>> # if value is not specified, zero-field is defined
         >>> field = df.Field(mesh=mesh, nvdim=3)
-        >>> field.value
-        0.0
-        >>> field.value = (0, 0, 1)
-        >>> field.value
-        (0, 0, 1)
+        >>> field.mean()
+        array([0., 0., 0.])
+        >>> field.update_field_values((0, 0, 1))
+        >>> field.mean()
+        array([0., 0., 1.])
         >>> # Setting the field value using a Python function (callable).
         >>> def value_function(point):
         ...     x, y, z = point
         ...     if x <= 1:
         ...         return (0, 0, 1)
         ...     else:
         ...         return (0, 0, -1)
-        >>> field.value = value_function
-        >>> field.value
-        <function value_function at ...>
-        >>> # We now change the value of a single cell so that the
-        >>> # representation used for initialising field is not valid
-        >>> # anymore.
-        >>> field.array[0, 0, 0, :] = (0, 0, 0)
-        >>> field.value
-        array(...)
-        >>> field.value.shape
-        (2, 2, 1, 3)
+        >>> field.update_field_values(value_function)
+        >>> field.array[0, 0, 0, :]
+        array([0., 0., 1.])
+        >>> field.array[1, 0, 0, :]
+        array([ 0.,  0., -1.])
```

Comment:
Would it be better to use `__call__` instead of directly accessing the numpy array?